interceptors/validator/validator.go

```diff
@@ -50,78 +63,60 @@ func validate(req any, all bool) error {
 	switch v := req.(type) {
 	case validatorLegacy:
 		if err := v.Validate(); err != nil {
+			log(level, logger, err.Error())
 			return status.Error(codes.InvalidArgument, err.Error())
 		}
 	case validator:
 		if err := v.Validate(false); err != nil {
+			log(level, logger, err.Error())
 			return status.Error(codes.InvalidArgument, err.Error())
 		}
 	}
 	return nil
 }
 
-// UnaryServerInterceptor returns a new unary server interceptor that validates incoming messages.
-//
-// Invalid messages will be rejected with `InvalidArgument` before reaching any userspace handlers.
-// If `all` is false, the interceptor returns first validation error. Otherwise, the interceptor
-// returns ALL validation error as a wrapped multi-error.
-// Note that generated codes prior to protoc-gen-validate v0.6.0 do not provide an all-validation
-// interface. In this case the interceptor fallbacks to legacy validation and `all` is ignored.
-func UnaryServerInterceptor(all bool) grpc.UnaryServerInterceptor {
-	return func(ctx context.Context, req any, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (any, error) {
-		if err := validate(req, all); err != nil {
+func UnaryServerInterceptor(opts ...Option) grpc.UnaryServerInterceptor {
+	o := evaluateServerOpt(opts)
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		if err := validate(req, o.shouldFailFast, o.logger); err != nil {
 			return nil, err
 		}
 		return handler(ctx, req)
 	}
 }
 
-// UnaryClientInterceptor returns a new unary client interceptor that validates outgoing messages.
-//
-// Invalid messages will be rejected with `InvalidArgument` before sending the request to server.
-// If `all` is false, the interceptor returns first validation error. Otherwise, the interceptor
-// returns ALL validation error as a wrapped multi-error.
-// Note that generated codes prior to protoc-gen-validate v0.6.0 do not provide an all-validation
-// interface. In this case the interceptor fallbacks to legacy validation and `all` is ignored.
-func UnaryClientInterceptor(all bool) grpc.UnaryClientInterceptor {
-	return func(ctx context.Context, method string, req, reply any, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
-		if err := validate(req, all); err != nil {
+func UnaryClientInterceptor(opts ...Option) grpc.UnaryClientInterceptor {
+	o := evaluateClientOpt(opts)
+	return func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+		if err := validate(req, o.shouldFailFast, o.logger); err != nil {
 			return err
 		}
 		return invoker(ctx, method, req, reply, cc, opts...)
 	}
 }
 
-// StreamServerInterceptor returns a new streaming server interceptor that validates incoming messages.
```

Comment:
You can leave majority of this comment e.g especially protoc-gen-validate info still applies.

Reply:
updated the comments. also, slight modification in the older ones